src/node/rpc/frontend.h

```diff
@@ -622,6 +625,11 @@ namespace ccf
 
       history->clear_on_result();
 
+      if (!has_updated_merkle_root)
```

Comment:
when would this not have been updated?

Reply:
if there is no read or write set in the transaction